### DISCUSSION TITLE
FNIRSI GC-01 (CH32F103R8): Replacing HH614 tube with J321 when R38 is not used

### DISCUSSION THREAD

**Discussion by alfmck** [Gissio/radpro#61] [Show and tell]
After replacing H614 to J321 I noticed, that using Factory default HV set, my device is registering very high background levels (about 120 cpm). That was the same for original and radpro firmware's. I measured HV - it was very high. Luckily you can use Custom HV settings in Radpro (thank Gissio). I can confirm that J321 optimal HV is 440V. Custom setting pairs in my case are 2.5kHz@1%, 5kHz@1.5%, 10kHz@3% - the background level I get is about 24 cpm; when measuring 1.5 mkSv/h I get about 600cpm.
But for H614 HV=440V is too low - I got about 3pcm background level and about 90cpm, when measuring 1.5 mkSv/h. 
So setting higher HV in original firmware or Factory default or radPro when H614 used is acceptable.
But when replace to J321, you must set HV 440V before. I suggest that could be DIFFERENT FACTORY SETTINGS for H614 and J321 and changed when you set a conversion factor for exact tube.
BTW: I used newer version of radpro 2.0rc2 and noticed that settings/right switch has action delaying. I suppose that RELATIVELLY CONSTANT ZERO  for 1 msek on switch would be enough.
And about measuring HV: you should use voltmeter or oscilloscope with PROBE and  at least 1gOm resistor.

**Comment by Gissio** [comment]
My reasoning: if you change the tube, you supposedly know what you are doing. If you know what you are doing, you can set up a custom HV profile.

What do you mean by "the settings/right switch has action delaying"? Your next comment suggests you are discussing the way the keyboard press is detected. I'm doing something very simple: sampling the keyboard ports every 25 ms. This avoids bounce events and minimizes code and CPU usage.

**Reply by alfmck** [reply]
Great thanks GISSIO for 2.0rc3!
Thank for energy saving HV profile. It measures average 22cpm for background. More precise measurements I'll do next week.
And no delaying on SETTINGS/RIGHT button.

**Reply by Gissio** [reply]
Please help me, I don't understand. By "action delaying", do you mean the counter generated right/settings key presses as if you had pressed the key, but you didn't?

**Reply by alfmck** [reply]
Yes. I use only DOWN button for settings, and for example when I make PWM duty cycle setting where are many values, I press and hold the button, the setting process is going slowly and after releasing the key the counter generates extra key press.
But on 2.0rc3 everything is OK.

**Reply by Gissio** [reply]
So your right/settings key port is not broken. I'm glad.

I solved this by synchronizing the sampling of the key presses to the HV pulses.

**Reply by alfmck** [reply]
Unfortunately broken. Every time I have to make changes in src in order to program UP key for fulfilling RIGHT/SETTING key function. So for the choosing a value I have only DOWN key.

**Reply by alfmck** [reply]
I have an option to restore my SETUP/RIGHT key. As shown in GC-01 schematics, PC11, PC12 are not used. I can connect my SETUP/RIGHT key from broken PC1 to PC11 and make change in src.
In order to extend GC-01 measurement ranges I'm going to place an extra low sensity tube and switch to it in case high levels of radiation. Using PC11 or PC12, and placing relay for switching tubes, radpro could automatically make switching and setting GC-01 for second tube data.

**Reply by alfmck** [reply]
Synchronizing the sampling of the key presses to the HV pulses have to solve EMI keyboard activating issue. As I tried to show signals on the buttons in oscilloscope videos, tube ionization pulses have even bigger amplitude than HV pulses. The bad state is when HV pulses and tube ionization pulses come at the same time and are summarizing on the buttons.

**Reply by Gissio** [reply]
Can you tell us about your experience with the haptic pulses feature? Are haptic pulses too weak/too strong?

**Reply by alfmck** [reply]
When haptic set to strong, the feeling is common like in my iphone8

**Reply by Gissio** [reply]
And when weak?

**Reply by alfmck** [reply]
When weak I feel nothing

**Reply by Gissio** [reply]
So weak has to be increased. Thanks.
